pkg/proc/fncall.go

```diff
@@ -130,9 +138,9 @@ func (callCtx *callContext) doReturn(ret *Variable, err error) {
 // EvalExpressionWithCalls is like EvalExpression but allows function calls in 'expr'.
 // Because this can only be done in the current goroutine, unlike
 // EvalExpression, EvalExpressionWithCalls is not a method of EvalScope.
-func EvalExpressionWithCalls(p Process, g *G, expr string, retLoadCfg LoadConfig, checkEscape bool) error {
-	bi := p.BinInfo()
-	if !p.Common().fncallEnabled {
+func EvalExpressionWithCalls(t *Target, g *G, expr string, retLoadCfg LoadConfig, checkEscape bool) error {
+	bi := t.BinInfo()
+	if ok, _ := t.Recorded(); ok {
```

Comment:
I think we should keep fncallEnabled, even though we are currently badly failing at setting the correct value for it (it should be false for rr and for arm64). For two reasons:

1. Function calls can be disabled for backends other than recordings (core files, the arm64 backend)
2. It should be actually possible to make function calls on the 'rr' backend through a facility called 'excursion', it isn't properly documented and I couldn't get it to work, but it exists.

I think it can be a parameter to NewTarget.